test/jdk/java/lang/runtime/RuntimeExitLogTest.java

```diff
@@ -0,0 +1,110 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.ParameterizedTest;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/*
+ * @test
+ * @summary verify logging of call to System.exit or Runtime.exit.
+ * @run junit/othervm RuntimeExitLogTest
+ */
+
+public class RuntimeExitLogTest {
+
+    private static final String TEST_JDK = System.getProperty("test.jdk");
+    private static final String TEST_SRC = System.getProperty("test.src");
+
+    /**
+     * Call System.exit() with the parameter (or zero if not supplied).
+     * @param args zero or 1 argument, an exit status
+     */
+    public static void main(String[] args) throws InterruptedException {
+        int status = args.length > 0 ? Integer.parseInt(args[0]) : 0;
+        System.exit(status);
+    }
+
+    /**
+     * Test various log level settings, and none.
+     * @return a stream of arguments for parameterized test
+     */
+    private static Stream<Arguments> logParamProvider() {
+        return Stream.of(
+                Arguments.of("ExitLogging-FINE.properties", 1, true),
+                Arguments.of("ExitLogging-INFO.properties", 2, false),
+                Arguments.of(null, 3, false)
+        );
+    }
+
+    /**
+     * Check that the logger output of a launched process contains the expected message.
+     * @param logProps The name of the log.properties file to set on the command line
+     * @param status the expected exit status of the process
+     * @param shouldLog true if the log should contain the message expected from Runtime.exit(status)
+     */
+    @ParameterizedTest
+    @MethodSource("logParamProvider")
+    public void checkLogger(String logProps, int status, boolean shouldLog) {
+        ProcessBuilder pb = new ProcessBuilder();
+        pb.redirectErrorStream(true);
+
+        List<String> cmd = pb.command();
+        cmd.add(Path.of(TEST_JDK,"bin", "java").toString());
+        if (logProps != null) {
+            cmd.add("-Djava.util.logging.config.file=" + Path.of(TEST_SRC, logProps).toString());
+        }
+        cmd.add(this.getClass().getName());
+        cmd.add(Integer.toString(status));
```

Comment:
Another possibility for testing this is to launch with ` --limit-modules java.base -Djdk.system.logger.level=DEBUG` to use the simple console implementation that is in java.base and avoid needing properties files for j.u.logging. Just mentioning is an option to make it simple.

Reply:
Good point - though maybe both configurations should be tested. The j.u.l.LogManager registers some shutdown hook so I believe it's important to test with j.u.l present too.

Reply:
Added a test of the console config with --limit-modules as suggested.